elf/module_test.go

```diff
@@ -0,0 +1,33 @@
+// +build linux
+
+package elf
+
+import (
+	"testing"
+)
+
+func TestUpdateKprobeSecName(t *testing.T) {
```

Comment:
I think what we want to test is the side effect of this, not the `Name` field.

So basically we want to have a Kprobe handler under an arbitrary name `SEC("foobar")`, attach that handler to Kprobe by doing something like

```go
m.UpdateKprobeSecName("foobar,"kprobe/whatever")
m.EnableKprobe("foobar")
```
and finally verify that your handler was executed when `whatever` ran.

Reply:
I was thinking about the same thing, but the direct result of calling `EnableKprobe()` is not easy to check(requires mount of debug, etc), that's why the overall tests in elf is lacking.

Reply:
How about adding a new function `EnableKProbeForFunction(secName string, funcName string, isRetProbe bool, string maxactive)` that doesn't tie the function to the name of the entry in `b.probes` ?   You could implement `EnableKProbe` in terms of `EnableKProbeForFunction`.

Reply:
@leeavital there are other ways that `EnableKprobe` is used(https://github.com/DataDog/gobpf/blob/master/elf/module.go#L373), if you don't change the actual kprobe name, you might ended up with surprises for the inconsistencies.

Reply:
Won't `EnableKrpobes` break anyway? If there is a kprobe named `kprobe/udp_sendmsg/rhel`, the it will try to attach that kprobe to a function named `udp_sendmsg/rhel`  which won't exist.

Reply:
It won't break if you call "update" first, but it's hard to call both `EnableKprobe` and `EnableKProbeForFunction ` in a for loop in `EnableKprobes` if you want to enable certain kprobe differently.

Reply:
Looking at your other PR I'm starting to think that a cleaner way to do this would be not changing the API of the package at all and establishing another convention like:
```
SEC("kprobe/tcp_sendmsg/arbitrary_sufix")
```
which would generate a handler that will get attached `kprobe/tcp_sendmsg`. We would simply dismiss everything after the second `/`.